src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/Orchestrator/OrchestratorMethodCollector.cs

```diff
@@ -0,0 +1,83 @@
+                && SyntaxNodeUtils.IsInsideOrchestratorFunction(semanticModel, declaration)
+                && symbol != null)
+            {
+                var methodInformation = new MethodInformation()
+                {
+                    SemanticModel = semanticModel,
+                    Declaration = declaration,
+                    DeclarationSymbol = symbol,
+                };
+
+                if (!this.orchestratorMethodDeclarations.ContainsKey(symbol))
+                {
+                    this.orchestratorMethodDeclarations.Add(symbol, methodInformation);
+
+                    this.FindInvokedMethods(semanticModel, methodInformation);
+                }
+            }
+        }
+
+        private void FindInvokedMethods(SemanticModel semanticModel, MethodInformation parentMethodInformation)
+        {
+            var parentDeclaration = parentMethodInformation.Declaration;
+            var invocationExpressions = parentDeclaration.DescendantNodes().OfType<InvocationExpressionSyntax>();
+            foreach (var invocation in invocationExpressions)
+            {
+                if (SyntaxNodeUtils.TryGetDeclaredSyntaxNode(semanticModel, invocation, out SyntaxNode invokedMethodDeclaration)
+                    && invokedMethodDeclaration is MethodDeclarationSyntax
+                    && SyntaxNodeUtils.TryGetISymbol(semanticModel, invocation, out ISymbol invokedSymbol))
+                {
+
+                    if (this.orchestratorMethodDeclarations.TryGetValue(invokedSymbol, out MethodInformation existingMethodInformation))
+                    {
+                        existingMethodInformation.Invocations.Add(invocation);
+                        if (!existingMethodInformation.Equals(parentMethodInformation))
+                        {
+                            existingMethodInformation.Parents.Add(parentMethodInformation);
+                        }
+                    }
+                    else
+                    {
+                        var invokedMethodInformation = new MethodInformation()
+                        {
+                            SemanticModel = semanticModel,
+                            Declaration = invokedMethodDeclaration,
+                            DeclarationSymbol = invokedSymbol,
+                            Invocations = new List<InvocationExpressionSyntax>() { invocation },
+                            Parents = new HashSet<MethodInformation>(new List<MethodInformation>() { parentMethodInformation }),
+                        };
+
+                        this.orchestratorMethodDeclarations.Add(invokedSymbol, invokedMethodInformation);
+
+                        FindInvokedMethods(semanticModel, invokedMethodInformation);
+                    }
+                }
+            }
+        }
+
+        public IEnumerable<MethodInformation> GetOrchestratorMethods()
+        {
+            return orchestratorMethodDeclarations.Values.ToList();
```

Comment:
nit: orchestrationMethodDeclarations.Values is already of type `IEnumerable<MethodInformation>`